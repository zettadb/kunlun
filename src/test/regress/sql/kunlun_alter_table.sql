drop table if exists t1;
create table t1(a serial primary key, b int);
insert into t1(b) values(11),(22),(33);

alter table t1 add column c int not null;
\d+ t1;
select*from t1;
alter table t1 drop column c;
select*from t1;
alter table t1 add column c int;
\d+ t1;
select*from t1;



alter table t1 add column d varchar(32) not null;
alter table t1 add column d varchar(32);
\d+ t1;
select*, d is null as disnull, length(d) from t1;

alter table t1 alter column d set default 'ddd';
\d+ t1;
insert into t1(b) values(77);
select*from t1;
alter table t1 alter column d drop default;
\d+ t1;



create table t30(a serial primary key, b int unique not null default 3);
alter table t30 add column c serial unique;
insert into t30 (b) values(11),(12),(13);
insert into t30 (b) values(14);
insert into t30 (b) values(15);
insert into t30 (b) values(16);
insert into t30 (b) values(17);
select*from t30;

alter table t30 rename a to aa;
alter table t30 rename b to bb;
alter table t30 rename c to cc;

alter table t30 add column e int not null;
alter table t30 add column f int;
alter table t30 alter column bb drop default;
alter table t30 alter column bb add generated by default as identity(start 100);
\d+ t30;
insert into t30(e) values(1111),(1111),(1111),(1111);
update t30 set f=2222;
select*from t30;

alter table t30 alter column e add generated by default as identity, alter column f set not null, alter column f add generated by default as identity, alter bb drop identity;
\d+ t30;


drop table if exists t1;
drop table if exists t100;
drop table if exists t101;
drop table if exists t102;
drop table if exists t103;
drop table if exists t30;
drop table if exists t301;
drop table if exists t302;
drop table if exists t303;
drop table if exists t304;

create table t1(a serial primary key, b int) partition by hash(a);
create table t100 partition of t1 for values with (modulus 4, remainder 0);
create table t101 partition of t1 for values with (modulus 4, remainder 1);
create table t102 partition of t1 for values with (modulus 4, remainder 2);
create table t103 partition of t1 for values with (modulus 4, remainder 3);

insert into t1(b) values(11),(22),(33);

alter table t1 add column c int not null;

\d+ t1;
select*from t1;
alter table t1 drop column c;
select*from t1;

alter table t1 add column d varchar(32) not null;
alter table t1 add column d varchar(32);
\d+ t1;
select*, d is null as disnull, length(d) from t1;

alter table t1 alter column d set default 'ddd';
\d+ t1;
insert into t1(b) values(77);
select*from t1;
alter table t1 alter column d drop default;
\d+ t1;
insert into t1(b) values(77);
select*from t1;



create table t30(a serial , b int unique not null default 3, primary key(a,b)) partition by range(b);
create table t301 partition of t30 for values from (0) to (10000);
create table t302 partition of t30 for values from (10000) to (20000);
create table t303 partition of t30 for values from (20000) to (30000);
create table t304 partition of t30 for values from (30000) to (40000);

alter table t30 add column c serial;
insert into t30 (b) values(11),(12),(13),(14),(15);
insert into t30 (b) values(14000),(14001),(14002),(14003),(14004);
insert into t30 (b) values(25000),(25001),(25002),(25003),(25004);
insert into t30 (b) values(36000),(36001),(36002),(36003),(36004);
select*from t30;

alter table t30 add column d serial;
select*from t30;

alter table t30 rename a to aa;
alter table t30 rename b to bb;
alter table t30 rename c to cc;

alter table t30 add column e int not null;
alter table t30 add column f int;
alter table t30 alter column bb drop default, alter column bb add generated by default as identity(start 100);
\d+ t30;
insert into t30(e) values(1111),(1111),(1111),(1111);
update t30 set f=2222;
select*from t30;

alter table t30 alter column e add generated by default as identity, alter column f set not null, alter column f add generated by default as identity, alter bb drop identity;
\d+ t30;
insert into t30(bb) values(200),(201),(202),(203);
select*from t30;


