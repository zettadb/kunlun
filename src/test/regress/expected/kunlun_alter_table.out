drop table if exists t1;
psql:sql/kunlun_alter_table.sql:1: NOTICE:  table "t1" does not exist, skipping
DROP TABLE
create table t1(a serial primary key, b int);
CREATE TABLE
insert into t1(b) values(11),(22),(33);
INSERT 0 3
alter table t1 add column c int not null;
ALTER TABLE
\d+ t1;
                                                Table "public.t1"
 Column |  Type   | Collation | Nullable |             Default             | Storage | Stats target | Description 
--------+---------+-----------+----------+---------------------------------+---------+--------------+-------------
 a      | integer |           | not null | "nextval"('t1_a_seq'::regclass) | plain   |              | 
 b      | integer |           |          |                                 | plain   |              | 
 c      | integer |           | not null |                                 | plain   |              | 
Indexes:
    "t1_pkey" PRIMARY KEY, btree (a NULLS FIRST)
Options: shard=1

select*from t1;
 a | b  | c 
---+----+---
 1 | 11 | 0
 2 | 22 | 0
 3 | 33 | 0
(3 rows)

alter table t1 drop column c;
ALTER TABLE
select*from t1;
 a | b  
---+----
 1 | 11
 2 | 22
 3 | 33
(3 rows)

alter table t1 add column c int;
ALTER TABLE
\d+ t1;
                                                Table "public.t1"
 Column |  Type   | Collation | Nullable |             Default             | Storage | Stats target | Description 
--------+---------+-----------+----------+---------------------------------+---------+--------------+-------------
 a      | integer |           | not null | "nextval"('t1_a_seq'::regclass) | plain   |              | 
 b      | integer |           |          |                                 | plain   |              | 
 c      | integer |           |          |                                 | plain   |              | 
Indexes:
    "t1_pkey" PRIMARY KEY, btree (a NULLS FIRST)
Options: shard=1

select*from t1;
 a | b  | c 
---+----+---
 1 | 11 |  
 2 | 22 |  
 3 | 33 |  
(3 rows)

alter table t1 add column d varchar(32) not null;
ALTER TABLE
alter table t1 add column d varchar(32);
psql:sql/kunlun_alter_table.sql:17: ERROR:  column "d" of relation "t1" already exists
\d+ t1;
                                                        Table "public.t1"
 Column |         Type          | Collation | Nullable |             Default             | Storage  | Stats target | Description 
--------+-----------------------+-----------+----------+---------------------------------+----------+--------------+-------------
 a      | integer               |           | not null | "nextval"('t1_a_seq'::regclass) | plain    |              | 
 b      | integer               |           |          |                                 | plain    |              | 
 c      | integer               |           |          |                                 | plain    |              | 
 d      | character varying(32) |           | not null |                                 | extended |              | 
Indexes:
    "t1_pkey" PRIMARY KEY, btree (a NULLS FIRST)
Options: shard=1

select*, d is null as disnull, length(d) from t1;
 a | b  | c | d | disnull | length 
---+----+---+---+---------+--------
 1 | 11 |   |   | f       |      0
 2 | 22 |   |   | f       |      0
 3 | 33 |   |   | f       |      0
(3 rows)

alter table t1 alter column d set default 'ddd';
ALTER TABLE
\d+ t1;
                                                        Table "public.t1"
 Column |         Type          | Collation | Nullable |             Default             | Storage  | Stats target | Description 
--------+-----------------------+-----------+----------+---------------------------------+----------+--------------+-------------
 a      | integer               |           | not null | "nextval"('t1_a_seq'::regclass) | plain    |              | 
 b      | integer               |           |          |                                 | plain    |              | 
 c      | integer               |           |          |                                 | plain    |              | 
 d      | character varying(32) |           | not null | 'ddd'::character varying        | extended |              | 
Indexes:
    "t1_pkey" PRIMARY KEY, btree (a NULLS FIRST)
Options: shard=1

insert into t1(b) values(77);
INSERT 0 1
select*from t1;
 a | b  | c |  d  
---+----+---+-----
 1 | 11 |   | 
 2 | 22 |   | 
 3 | 33 |   | 
 4 | 77 |   | ddd
(4 rows)

alter table t1 alter column d drop default;
ALTER TABLE
\d+ t1;
                                                        Table "public.t1"
 Column |         Type          | Collation | Nullable |             Default             | Storage  | Stats target | Description 
--------+-----------------------+-----------+----------+---------------------------------+----------+--------------+-------------
 a      | integer               |           | not null | "nextval"('t1_a_seq'::regclass) | plain    |              | 
 b      | integer               |           |          |                                 | plain    |              | 
 c      | integer               |           |          |                                 | plain    |              | 
 d      | character varying(32) |           | not null |                                 | extended |              | 
Indexes:
    "t1_pkey" PRIMARY KEY, btree (a NULLS FIRST)
Options: shard=1

create table t30(a serial primary key, b int unique not null default 3);
CREATE TABLE
alter table t30 add column c serial unique;
psql:sql/kunlun_alter_table.sql:31: WARNING:  Sequence cannot be used until transaction commits
HINT:  If you are adding a serial column, make sure the original table is empty, as the new column will not be initialized by the sequence
ALTER TABLE
insert into t30 (b) values(11),(12),(13);
INSERT 0 3
insert into t30 (b) values(14);
INSERT 0 1
insert into t30 (b) values(15);
INSERT 0 1
insert into t30 (b) values(16);
INSERT 0 1
insert into t30 (b) values(17);
INSERT 0 1
select*from t30;
 a | b  | c 
---+----+---
 1 | 11 | 1
 2 | 12 | 2
 3 | 13 | 3
 4 | 14 | 4
 5 | 15 | 5
 6 | 16 | 6
 7 | 17 | 7
(7 rows)

alter table t30 rename a to aa;
ALTER TABLE
alter table t30 rename b to bb;
ALTER TABLE
alter table t30 rename c to cc;
ALTER TABLE
alter table t30 add column e int not null;
ALTER TABLE
alter table t30 add column f int;
ALTER TABLE
alter table t30 alter column bb drop default;
ALTER TABLE
alter table t30 alter column bb add generated by default as identity(start 100);
ALTER TABLE
\d+ t30;
                                                Table "public.t30"
 Column |  Type   | Collation | Nullable |             Default              | Storage | Stats target | Description 
--------+---------+-----------+----------+----------------------------------+---------+--------------+-------------
 aa     | integer |           | not null | "nextval"('t30_a_seq'::regclass) | plain   |              | 
 bb     | integer |           | not null | generated by default as identity | plain   |              | 
 cc     | integer |           | not null | "nextval"('t30_c_seq'::regclass) | plain   |              | 
 e      | integer |           | not null |                                  | plain   |              | 
 f      | integer |           |          |                                  | plain   |              | 
Indexes:
    "t30_pkey" PRIMARY KEY, btree (aa NULLS FIRST)
    "t30_b_key" UNIQUE CONSTRAINT, btree (bb NULLS FIRST)
    "t30_c_key" UNIQUE CONSTRAINT, btree (cc NULLS FIRST)
Options: shard=1

insert into t30(e) values(1111),(1111),(1111),(1111);
INSERT 0 4
update t30 set f=2222;
UPDATE 11
select*from t30;
 aa | bb  | cc |  e   |  f   
----+-----+----+------+------
  1 |  11 |  1 |    0 | 2222
  2 |  12 |  2 |    0 | 2222
  3 |  13 |  3 |    0 | 2222
  4 |  14 |  4 |    0 | 2222
  5 |  15 |  5 |    0 | 2222
  6 |  16 |  6 |    0 | 2222
  7 |  17 |  7 |    0 | 2222
  8 | 100 |  8 | 1111 | 2222
  9 | 101 |  9 | 1111 | 2222
 10 | 102 | 10 | 1111 | 2222
 11 | 103 | 11 | 1111 | 2222
(11 rows)

alter table t30 alter column e add generated by default as identity, alter column f set not null, alter column f add generated by default as identity, alter bb drop identity;
ALTER TABLE
\d+ t30;
                                                Table "public.t30"
 Column |  Type   | Collation | Nullable |             Default              | Storage | Stats target | Description 
--------+---------+-----------+----------+----------------------------------+---------+--------------+-------------
 aa     | integer |           | not null | "nextval"('t30_a_seq'::regclass) | plain   |              | 
 bb     | integer |           | not null |                                  | plain   |              | 
 cc     | integer |           | not null | "nextval"('t30_c_seq'::regclass) | plain   |              | 
 e      | integer |           | not null | generated by default as identity | plain   |              | 
 f      | integer |           | not null | generated by default as identity | plain   |              | 
Indexes:
    "t30_pkey" PRIMARY KEY, btree (aa NULLS FIRST)
    "t30_b_key" UNIQUE CONSTRAINT, btree (bb NULLS FIRST)
    "t30_c_key" UNIQUE CONSTRAINT, btree (cc NULLS FIRST)
Options: shard=1

drop table if exists t1;
DROP TABLE
drop table if exists t100;
psql:sql/kunlun_alter_table.sql:57: NOTICE:  table "t100" does not exist, skipping
DROP TABLE
drop table if exists t101;
psql:sql/kunlun_alter_table.sql:58: NOTICE:  table "t101" does not exist, skipping
DROP TABLE
drop table if exists t102;
psql:sql/kunlun_alter_table.sql:59: NOTICE:  table "t102" does not exist, skipping
DROP TABLE
drop table if exists t103;
psql:sql/kunlun_alter_table.sql:60: NOTICE:  table "t103" does not exist, skipping
DROP TABLE
drop table if exists t30;
DROP TABLE
drop table if exists t301;
psql:sql/kunlun_alter_table.sql:62: NOTICE:  table "t301" does not exist, skipping
DROP TABLE
drop table if exists t302;
psql:sql/kunlun_alter_table.sql:63: NOTICE:  table "t302" does not exist, skipping
DROP TABLE
drop table if exists t303;
psql:sql/kunlun_alter_table.sql:64: NOTICE:  table "t303" does not exist, skipping
DROP TABLE
drop table if exists t304;
psql:sql/kunlun_alter_table.sql:65: NOTICE:  table "t304" does not exist, skipping
DROP TABLE
create table t1(a serial primary key, b int) partition by hash(a);
CREATE TABLE
create table t100 partition of t1 for values with (modulus 4, remainder 0);
CREATE TABLE
create table t101 partition of t1 for values with (modulus 4, remainder 1);
CREATE TABLE
create table t102 partition of t1 for values with (modulus 4, remainder 2);
CREATE TABLE
create table t103 partition of t1 for values with (modulus 4, remainder 3);
CREATE TABLE
insert into t1(b) values(11),(22),(33);
INSERT 0 3
alter table t1 add column c int not null;
ALTER TABLE
\d+ t1;
                                          Partitioned table "public.t1"
 Column |  Type   | Collation | Nullable |             Default             | Storage | Stats target | Description 
--------+---------+-----------+----------+---------------------------------+---------+--------------+-------------
 a      | integer |           | not null | "nextval"('t1_a_seq'::regclass) | plain   |              | 
 b      | integer |           |          |                                 | plain   |              | 
 c      | integer |           | not null |                                 | plain   |              | 
Partition key: HASH (a)
Indexes:
    "t1_pkey" PRIMARY KEY, btree (a)
Partitions: t100 FOR VALUES WITH (modulus 4, remainder 0),
            t101 FOR VALUES WITH (modulus 4, remainder 1),
            t102 FOR VALUES WITH (modulus 4, remainder 2),
            t103 FOR VALUES WITH (modulus 4, remainder 3)

select*from t1;
 a | b  | c 
---+----+---
 1 | 11 | 0
 3 | 33 | 0
 2 | 22 | 0
(3 rows)

alter table t1 drop column c;
ALTER TABLE
select*from t1;
 a | b  
---+----
 1 | 11
 3 | 33
 2 | 22
(3 rows)

alter table t1 add column d varchar(32) not null;
ALTER TABLE
alter table t1 add column d varchar(32);
psql:sql/kunlun_alter_table.sql:83: ERROR:  column "d" of relation "t1" already exists
\d+ t1;
                                                  Partitioned table "public.t1"
 Column |         Type          | Collation | Nullable |             Default             | Storage  | Stats target | Description 
--------+-----------------------+-----------+----------+---------------------------------+----------+--------------+-------------
 a      | integer               |           | not null | "nextval"('t1_a_seq'::regclass) | plain    |              | 
 b      | integer               |           |          |                                 | plain    |              | 
 d      | character varying(32) |           | not null |                                 | extended |              | 
Partition key: HASH (a)
Indexes:
    "t1_pkey" PRIMARY KEY, btree (a)
Partitions: t100 FOR VALUES WITH (modulus 4, remainder 0),
            t101 FOR VALUES WITH (modulus 4, remainder 1),
            t102 FOR VALUES WITH (modulus 4, remainder 2),
            t103 FOR VALUES WITH (modulus 4, remainder 3)

select*, d is null as disnull, length(d) from t1;
 a | b  | d | disnull | length 
---+----+---+---------+--------
 1 | 11 |   | f       |      0
 3 | 33 |   | f       |      0
 2 | 22 |   | f       |      0
(3 rows)

alter table t1 alter column d set default 'ddd';
ALTER TABLE
\d+ t1;
                                                  Partitioned table "public.t1"
 Column |         Type          | Collation | Nullable |             Default             | Storage  | Stats target | Description 
--------+-----------------------+-----------+----------+---------------------------------+----------+--------------+-------------
 a      | integer               |           | not null | "nextval"('t1_a_seq'::regclass) | plain    |              | 
 b      | integer               |           |          |                                 | plain    |              | 
 d      | character varying(32) |           | not null | 'ddd'::character varying        | extended |              | 
Partition key: HASH (a)
Indexes:
    "t1_pkey" PRIMARY KEY, btree (a)
Partitions: t100 FOR VALUES WITH (modulus 4, remainder 0),
            t101 FOR VALUES WITH (modulus 4, remainder 1),
            t102 FOR VALUES WITH (modulus 4, remainder 2),
            t103 FOR VALUES WITH (modulus 4, remainder 3)

insert into t1(b) values(77);
INSERT 0 1
select*from t1;
 a | b  |  d  
---+----+-----
 1 | 11 | 
 3 | 33 | 
 2 | 22 | 
 4 | 77 | ddd
(4 rows)

alter table t1 alter column d drop default;
ALTER TABLE
\d+ t1;
                                                  Partitioned table "public.t1"
 Column |         Type          | Collation | Nullable |             Default             | Storage  | Stats target | Description 
--------+-----------------------+-----------+----------+---------------------------------+----------+--------------+-------------
 a      | integer               |           | not null | "nextval"('t1_a_seq'::regclass) | plain    |              | 
 b      | integer               |           |          |                                 | plain    |              | 
 d      | character varying(32) |           | not null |                                 | extended |              | 
Partition key: HASH (a)
Indexes:
    "t1_pkey" PRIMARY KEY, btree (a)
Partitions: t100 FOR VALUES WITH (modulus 4, remainder 0),
            t101 FOR VALUES WITH (modulus 4, remainder 1),
            t102 FOR VALUES WITH (modulus 4, remainder 2),
            t103 FOR VALUES WITH (modulus 4, remainder 3)

insert into t1(b) values(77);
psql:sql/kunlun_alter_table.sql:93: ERROR:  null value in column "d" violates not-null constraint
DETAIL:  Failing row contains (5, 77, null).
select*from t1;
 a | b  |  d  
---+----+-----
 1 | 11 | 
 3 | 33 | 
 2 | 22 | 
 4 | 77 | ddd
(4 rows)

create table t30(a serial , b int unique not null default 3, primary key(a,b)) partition by range(b);
CREATE TABLE
create table t301 partition of t30 for values from (0) to (10000);
CREATE TABLE
create table t302 partition of t30 for values from (10000) to (20000);
CREATE TABLE
create table t303 partition of t30 for values from (20000) to (30000);
CREATE TABLE
create table t304 partition of t30 for values from (30000) to (40000);
CREATE TABLE
alter table t30 add column c serial;
psql:sql/kunlun_alter_table.sql:104: WARNING:  Sequence cannot be used until transaction commits
HINT:  If you are adding a serial column, make sure the original table is empty, as the new column will not be initialized by the sequence
psql:sql/kunlun_alter_table.sql:104: WARNING:  Sequence cannot be used until transaction commits
HINT:  If you are adding a serial column, make sure the original table is empty, as the new column will not be initialized by the sequence
psql:sql/kunlun_alter_table.sql:104: WARNING:  Sequence cannot be used until transaction commits
HINT:  If you are adding a serial column, make sure the original table is empty, as the new column will not be initialized by the sequence
psql:sql/kunlun_alter_table.sql:104: WARNING:  Sequence cannot be used until transaction commits
HINT:  If you are adding a serial column, make sure the original table is empty, as the new column will not be initialized by the sequence
ALTER TABLE
insert into t30 (b) values(11),(12),(13),(14),(15);
INSERT 0 5
insert into t30 (b) values(14000),(14001),(14002),(14003),(14004);
INSERT 0 5
insert into t30 (b) values(25000),(25001),(25002),(25003),(25004);
INSERT 0 5
insert into t30 (b) values(36000),(36001),(36002),(36003),(36004);
INSERT 0 5
select*from t30;
 a  |   b   | c  
----+-------+----
  1 |    11 |  1
  2 |    12 |  2
  3 |    13 |  3
  4 |    14 |  4
  5 |    15 |  5
  6 | 14000 |  6
  7 | 14001 |  7
  8 | 14002 |  8
  9 | 14003 |  9
 10 | 14004 | 10
 11 | 25000 | 11
 12 | 25001 | 12
 13 | 25002 | 13
 14 | 25003 | 14
 15 | 25004 | 15
 16 | 36000 | 16
 17 | 36001 | 17
 18 | 36002 | 18
 19 | 36003 | 19
 20 | 36004 | 20
(20 rows)

alter table t30 add column d serial;
psql:sql/kunlun_alter_table.sql:111: WARNING:  Sequence cannot be used until transaction commits
HINT:  If you are adding a serial column, make sure the original table is empty, as the new column will not be initialized by the sequence
psql:sql/kunlun_alter_table.sql:111: WARNING:  Sequence cannot be used until transaction commits
HINT:  If you are adding a serial column, make sure the original table is empty, as the new column will not be initialized by the sequence
psql:sql/kunlun_alter_table.sql:111: WARNING:  Sequence cannot be used until transaction commits
HINT:  If you are adding a serial column, make sure the original table is empty, as the new column will not be initialized by the sequence
psql:sql/kunlun_alter_table.sql:111: WARNING:  Sequence cannot be used until transaction commits
HINT:  If you are adding a serial column, make sure the original table is empty, as the new column will not be initialized by the sequence
ALTER TABLE
select*from t30;
 a  |   b   | c  | d 
----+-------+----+---
  1 |    11 |  1 | 0
  2 |    12 |  2 | 0
  3 |    13 |  3 | 0
  4 |    14 |  4 | 0
  5 |    15 |  5 | 0
  6 | 14000 |  6 | 0
  7 | 14001 |  7 | 0
  8 | 14002 |  8 | 0
  9 | 14003 |  9 | 0
 10 | 14004 | 10 | 0
 11 | 25000 | 11 | 0
 12 | 25001 | 12 | 0
 13 | 25002 | 13 | 0
 14 | 25003 | 14 | 0
 15 | 25004 | 15 | 0
 16 | 36000 | 16 | 0
 17 | 36001 | 17 | 0
 18 | 36002 | 18 | 0
 19 | 36003 | 19 | 0
 20 | 36004 | 20 | 0
(20 rows)

alter table t30 rename a to aa;
ALTER TABLE
alter table t30 rename b to bb;
ALTER TABLE
alter table t30 rename c to cc;
ALTER TABLE
alter table t30 add column e int not null;
ALTER TABLE
alter table t30 add column f int;
ALTER TABLE
alter table t30 alter column bb drop default, alter column bb add generated by default as identity(start 100);
ALTER TABLE
\d+ t30;
                                          Partitioned table "public.t30"
 Column |  Type   | Collation | Nullable |             Default              | Storage | Stats target | Description 
--------+---------+-----------+----------+----------------------------------+---------+--------------+-------------
 aa     | integer |           | not null | "nextval"('t30_a_seq'::regclass) | plain   |              | 
 bb     | integer |           | not null | generated by default as identity | plain   |              | 
 cc     | integer |           | not null | "nextval"('t30_c_seq'::regclass) | plain   |              | 
 d      | integer |           | not null | "nextval"('t30_d_seq'::regclass) | plain   |              | 
 e      | integer |           | not null |                                  | plain   |              | 
 f      | integer |           |          |                                  | plain   |              | 
Partition key: RANGE (bb)
Indexes:
    "t30_pkey" PRIMARY KEY, btree (aa, bb)
    "t30_b_key" UNIQUE CONSTRAINT, btree (bb)
Partitions: t301 FOR VALUES FROM (0) TO (10000),
            t302 FOR VALUES FROM (10000) TO (20000),
            t303 FOR VALUES FROM (20000) TO (30000),
            t304 FOR VALUES FROM (30000) TO (40000)

insert into t30(e) values(1111),(1111),(1111),(1111);
INSERT 0 4
update t30 set f=2222;
UPDATE 24
select*from t30;
 aa |  bb   | cc | d |  e   |  f   
----+-------+----+---+------+------
  1 |    11 |  1 | 0 |    0 | 2222
  2 |    12 |  2 | 0 |    0 | 2222
  3 |    13 |  3 | 0 |    0 | 2222
  4 |    14 |  4 | 0 |    0 | 2222
  5 |    15 |  5 | 0 |    0 | 2222
 21 |   100 | 21 | 1 | 1111 | 2222
 22 |   101 | 22 | 2 | 1111 | 2222
 23 |   102 | 23 | 3 | 1111 | 2222
 24 |   103 | 24 | 4 | 1111 | 2222
  6 | 14000 |  6 | 0 |    0 | 2222
  7 | 14001 |  7 | 0 |    0 | 2222
  8 | 14002 |  8 | 0 |    0 | 2222
  9 | 14003 |  9 | 0 |    0 | 2222
 10 | 14004 | 10 | 0 |    0 | 2222
 11 | 25000 | 11 | 0 |    0 | 2222
 12 | 25001 | 12 | 0 |    0 | 2222
 13 | 25002 | 13 | 0 |    0 | 2222
 14 | 25003 | 14 | 0 |    0 | 2222
 15 | 25004 | 15 | 0 |    0 | 2222
 16 | 36000 | 16 | 0 |    0 | 2222
 17 | 36001 | 17 | 0 |    0 | 2222
 18 | 36002 | 18 | 0 |    0 | 2222
 19 | 36003 | 19 | 0 |    0 | 2222
 20 | 36004 | 20 | 0 |    0 | 2222
(24 rows)

alter table t30 alter column e add generated by default as identity, alter column f set not null, alter column f add generated by default as identity, alter bb drop identity;
ALTER TABLE
\d+ t30;
                                          Partitioned table "public.t30"
 Column |  Type   | Collation | Nullable |             Default              | Storage | Stats target | Description 
--------+---------+-----------+----------+----------------------------------+---------+--------------+-------------
 aa     | integer |           | not null | "nextval"('t30_a_seq'::regclass) | plain   |              | 
 bb     | integer |           | not null |                                  | plain   |              | 
 cc     | integer |           | not null | "nextval"('t30_c_seq'::regclass) | plain   |              | 
 d      | integer |           | not null | "nextval"('t30_d_seq'::regclass) | plain   |              | 
 e      | integer |           | not null | generated by default as identity | plain   |              | 
 f      | integer |           | not null | generated by default as identity | plain   |              | 
Partition key: RANGE (bb)
Indexes:
    "t30_pkey" PRIMARY KEY, btree (aa, bb)
    "t30_b_key" UNIQUE CONSTRAINT, btree (bb)
Partitions: t301 FOR VALUES FROM (0) TO (10000),
            t302 FOR VALUES FROM (10000) TO (20000),
            t303 FOR VALUES FROM (20000) TO (30000),
            t304 FOR VALUES FROM (30000) TO (40000)

insert into t30(bb) values(200),(201),(202),(203);
INSERT 0 4
select*from t30;
 aa |  bb   | cc | d |  e   |  f   
----+-------+----+---+------+------
  1 |    11 |  1 | 0 |    0 | 2222
  2 |    12 |  2 | 0 |    0 | 2222
  3 |    13 |  3 | 0 |    0 | 2222
  4 |    14 |  4 | 0 |    0 | 2222
  5 |    15 |  5 | 0 |    0 | 2222
 21 |   100 | 21 | 1 | 1111 | 2222
 22 |   101 | 22 | 2 | 1111 | 2222
 23 |   102 | 23 | 3 | 1111 | 2222
 24 |   103 | 24 | 4 | 1111 | 2222
 25 |   200 | 25 | 5 |    1 |    1
 26 |   201 | 26 | 6 |    2 |    2
 27 |   202 | 27 | 7 |    3 |    3
 28 |   203 | 28 | 8 |    4 |    4
  6 | 14000 |  6 | 0 |    0 | 2222
  7 | 14001 |  7 | 0 |    0 | 2222
  8 | 14002 |  8 | 0 |    0 | 2222
  9 | 14003 |  9 | 0 |    0 | 2222
 10 | 14004 | 10 | 0 |    0 | 2222
 11 | 25000 | 11 | 0 |    0 | 2222
 12 | 25001 | 12 | 0 |    0 | 2222
 13 | 25002 | 13 | 0 |    0 | 2222
 14 | 25003 | 14 | 0 |    0 | 2222
 15 | 25004 | 15 | 0 |    0 | 2222
 16 | 36000 | 16 | 0 |    0 | 2222
 17 | 36001 | 17 | 0 |    0 | 2222
 18 | 36002 | 18 | 0 |    0 | 2222
 19 | 36003 | 19 | 0 |    0 | 2222
 20 | 36004 | 20 | 0 |    0 | 2222
(28 rows)

